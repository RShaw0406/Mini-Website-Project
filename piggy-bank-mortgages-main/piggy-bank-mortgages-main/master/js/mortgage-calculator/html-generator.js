// Contains functionality that is used to render the correct HTMl sections

export const renderHtml = (section, applicantNo, applicant1Employed, applicant2Employed) => {
  // Handles which html rendering function to call depending on the section
  switch (section) {
    case 1:
      document.getElementById('personal-details').innerHTML = getPersonalDetails(applicantNo);
      break;
    case 3:
      document.getElementById('employment').innerHTML = getEmploymentDetails(applicantNo);
      break;
    case 4:
      document.getElementById('income').innerHTML = getIncomeDetails(
        applicantNo,
        applicant1Employed,
        applicant2Employed,
      );
      break;
    case 5:
      document.getElementById('outgoings').innerHTML = getOutgoingsDetails(applicantNo);
      break;
  }
};

const getPersonalDetails = (count) => {
  // Returns the personalDetails html section for each applicant
  let result = ``;
  for (let i = 1; i <= count; i++) {
    result += `<h4>Applicant ${i}:</h4>
        <div class="row pt-2">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName1">First name</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" placeholder="Enter first name"
                    name="firstname${i}" required>
            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName2">Surname</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" name="surname${i}"
                    placeholder="Enter surname" required>

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Email</label>
            <div class="col-sm-8 pt-1">
                <input type="email" class="form-control personal" name="email${i}"
                    placeholder="Enter email address" required>

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Phone number</label>
            <div class="col-sm-8 pt-1">
                <input type="number" class="form-control"
                                    placeholder="Enter mobile number(preferred)" name="phone" required>
            </div>
        </div>
    </div>
    </br>
    `;
  }
  return result;
};

const getEmploymentDetails = (count) => {
  // Returns the employmentDetails html section for each applicant
  let result = ``;
  for (let i = 1; i <= count; i++) {
    result += `<h4>Applicant ${i}:</h4>
        <div class="row">
        <h7 class="required">Are you employed?</h7>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed${i}" id="employed${i}" required>
        <label class="form-check-label" for="employed${i}">
            Yes
        </label>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed${i}" id="unemployed${i}" required>
        <label class="form-check-label" for="unemployed${i}">
            No
        </label>
        </div>
        </br>
        `;
  }
  return result;
};

const reducedIncomeHtml = `<div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the gross pay you receive, including benefits</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required>
                                </div>
                            </div>
                        </div>
                         </br>
                        `;

const employedIncomeHtml = `<div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is your gross yearly salary you receive from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required>
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much in bonuses do you receive per year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the total amount you receive as a bonus from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="bonus" step="any" required>
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">Are you paid anything else? if so tell us how much</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is any additional income you receive</small></p>
                                    <input type="number" class="form-control income" name="other" step="any" required>
                                </div>
                            </div>
                        </div>
                         </br>
                        `;

const getIncomeDetails = (count, applicant1Employed, applicant2Employed) => {
  // Returns the incomeDetails html section depending on whether each applicant is employed
  let result = ``;
  for (let i = 1; i <= count; i++) {
    result += `<h4>Applicant ${i}:</h4>`;
    if (i === 1) {
      if (applicant1Employed === true) {
        result += employedIncomeHtml;
      } else {
        result += reducedIncomeHtml;
      }
    } else {
      if (applicant2Employed === true) {
        result += employedIncomeHtml;
      } else {
        result += reducedIncomeHtml;
      }
    }
  }
  return result;
};

const getOutgoingsDetails = (count) => {
  // Returns the outgoingsDetails html section for each applicant
  let result = ``;
  for (let i = 1; i <= count; i++) {
    result += `<h4>Applicant ${i}:</h4>
                    <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required">How much are you paying off in credit cards?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is the total balance you currently hold on all credit cards</small>
                            </p>
                            <input type="number" class="form-control outgoings" name="credit-debt" step="any" required>
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much are you paying off in loans?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is for any outstanding loans e.g. car finance</small></p>
                            <input type="number" class="form-control outgoings" name="loan-debt" step="any" required>
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much in other expenses are you paying?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This could be for household bills such as electricity or gas</small></p>
                            <input type="number" class="form-control outgoings" name="other-debt" step="any" required>
                        </div>
                    </div>
                </div>
                </br>
                `;
  }
  return result;
};

const formatter = new Intl.NumberFormat('en-UK', {
  // Used to format number values into UK currency
  style: 'currency',
  currency: 'GBP',
});

export const getResults = (amount) => {
  // Returns the results html section containing the mortgage amount that can be borrowed
  if (amount > 0) {
    return `<div class="d-flex justify-content-center">
                    <div class="row pt-2">
                        <h3>Good news! you can borrow up to</h3>
                    </div>
                </div>
                <div class="d-flex justify-content-center">
                    <div class="row pt-2">
                        <h3>${formatter.format(amount)}</h3>
                    </div>
                </div>
        `;
  } else {
    return `<div class="d-flex justify-content-center">
                    <div class="row pt-2" align="center">
                        <h3>Sorry, we are unable to provide you with a mortgage based on the details you have provided</h3>
                    </div>
                </div>
        `;
  }
};

export const displayFailureMessage = (failMessage, form) => {
  failMessage.classList.remove('d-none');
  form.classList.add('was-validated');
}

