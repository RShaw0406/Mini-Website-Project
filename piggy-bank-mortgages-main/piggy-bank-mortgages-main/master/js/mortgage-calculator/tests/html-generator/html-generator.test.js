import {displayFailureMessage, getResults, renderHtml} from "../../html-generator";

describe('HTML Generator Tests', () => {
    describe('When calling renderHtml with a 1', () => {
        it('Should render the personalDetails HTML section correctly when called with a single applicant', () => {
            // Arrange
            document.body.innerHTML = '<div id="personal-details"></div>';
            // Act
            renderHtml(1, 1, true, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the personalDetails HTML section correctly when called with two applicants', () => {
            // Arrange
            document.body.innerHTML = '<div id="personal-details"></div>';
            // Act
            renderHtml(1, 2, false, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
    })
    describe('When calling renderHtml with a 3', () => {
        it('Should render the employmentDetails HTML section correctly when called with a single applicant', () => {
            // Arrange
            document.body.innerHTML = '<div id="employment"></div>';
            // Act
            renderHtml(3, 1, true, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the employmentDetails HTML section correctly when called with two applicants', () => {
            // Arrange
            document.body.innerHTML = '<div id="employment"></div>';
            // Act
            renderHtml(3, 2, false, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
    })
    describe('When calling renderHtml with a 4', () => {
        it('Should render the incomeDetails HTML section correctly when called with a single employed applicant', () => {
            // Arrange
            document.body.innerHTML = '<div id="income"></div>';
            // Act
            renderHtml(4, 1, true, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the incomeDetails HTML section correctly when called with a single unemployed applicant', () => {
            // Arrange
            document.body.innerHTML = '<div id="income"></div>';
            // Act
            renderHtml(4, 1, false, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the incomeDetails HTML section correctly when called with two employed applicants', () => {
            // Arrange
            document.body.innerHTML = '<div id="income"></div>';
            // Act
            renderHtml(4, 2, true, true)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the incomeDetails HTML section correctly when called with two unemployed applicants', () => {
            // Arrange
            document.body.innerHTML = '<div id="income"></div>';
            // Act
            renderHtml(4, 2, false, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the incomeDetails HTML section correctly when called with applicant1 employed and applicant2 unemployed', () => {
            // Arrange
            document.body.innerHTML = '<div id="income"></div>';
            // Act
            renderHtml(4, 2, true, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the incomeDetails HTML section correctly when called with applicant1 unemployed and applicant2 employed', () => {
            // Arrange
            document.body.innerHTML = '<div id="income"></div>';
            // Act
            renderHtml(4, 2, false, true)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
    })
    describe('When calling renderHtml with a 5', () => {
        it('Should render the outgoingsDetails HTML section correctly when called with a single applicant', () => {
            // Arrange
            document.body.innerHTML = '<div id="outgoings"></div>';
            // Act
            renderHtml(5, 1, true, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
        it('Should render the outgoingsDetails HTML section correctly when called with two applicants', () => {
            // Arrange
            document.body.innerHTML = '<div id="outgoings"></div>';
            // Act
            renderHtml(5, 2, false, false)
            // Assert
            expect(document.body.innerHTML).toMatchSnapshot();
        })
    })
    describe('When calling getResults with a valid amount', () => {
        it('Should render the results section correctly', () => {
            // Act
            const result = getResults(200000)
            // Assert
            expect(result).toMatchSnapshot()
        })
    })
    describe('When calling getResults with an invalid amount', () => {
        it('Should render the results section correctly', () => {
            // Act
            const result = getResults(0)
            // Assert
            expect(result).toMatchSnapshot()
        })
    })

    describe('When calling displayFailureMessage', () => {
        it('Should update the relevant classLists correctly', () => {
            // Arrange
            const failMessage = document.createElement('fail-message')
            failMessage.classList.add('d-none')
            const form = document.createElement('form')
            // Act
            displayFailureMessage(failMessage, form)
            // Assert
            expect([...failMessage.classList.values()]).toEqual([])
            expect([...form.classList.values()]).toEqual(['was-validated'])
        })
    })
})
