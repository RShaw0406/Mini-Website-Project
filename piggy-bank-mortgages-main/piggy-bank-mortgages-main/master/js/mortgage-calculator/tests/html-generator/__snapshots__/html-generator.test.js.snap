// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML Generator Tests When calling getResults with a valid amount Should render the results section correctly 1`] = `
"<div class="d-flex justify-content-center">
                    <div class="row pt-2">
                        <h3>Good news! you can borrow up to</h3>
                    </div>
                </div>
                <div class="d-flex justify-content-center">
                    <div class="row pt-2">
                        <h3>Â£200,000.00</h3>
                    </div>
                </div>
        "
`;

exports[`HTML Generator Tests When calling getResults with an invalid amount Should render the results section correctly 1`] = `
"<div class="d-flex justify-content-center">
                    <div class="row pt-2" align="center">
                        <h3>Sorry, we are unable to provide you with a mortgage based on the details you have provided</h3>
                    </div>
                </div>
        "
`;

exports[`HTML Generator Tests When calling renderHtml with a 1 Should render the personalDetails HTML section correctly when called with a single applicant 1`] = `
"<div id="personal-details"><h4>Applicant 1:</h4>
        <div class="row pt-2">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName1">First name</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" placeholder="Enter first name" name="firstname1" required="">
            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName2">Surname</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" name="surname1" placeholder="Enter surname" required="">

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Email</label>
            <div class="col-sm-8 pt-1">
                <input type="email" class="form-control personal" name="email1" placeholder="Enter email address" required="">

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Phone number</label>
            <div class="col-sm-8 pt-1">
                <input type="number" class="form-control" placeholder="Enter mobile number(preferred)" name="phone" required="">
            </div>
        </div>
    </div>
    <br>
    </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 1 Should render the personalDetails HTML section correctly when called with two applicants 1`] = `
"<div id="personal-details"><h4>Applicant 1:</h4>
        <div class="row pt-2">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName1">First name</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" placeholder="Enter first name" name="firstname1" required="">
            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName2">Surname</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" name="surname1" placeholder="Enter surname" required="">

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Email</label>
            <div class="col-sm-8 pt-1">
                <input type="email" class="form-control personal" name="email1" placeholder="Enter email address" required="">

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Phone number</label>
            <div class="col-sm-8 pt-1">
                <input type="number" class="form-control" placeholder="Enter mobile number(preferred)" name="phone" required="">
            </div>
        </div>
    </div>
    <br>
    <h4>Applicant 2:</h4>
        <div class="row pt-2">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName1">First name</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" placeholder="Enter first name" name="firstname2" required="">
            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputName2">Surname</label>
            <div class="col-sm-8 pt-1">
                <input type="text" class="form-control personal" name="surname2" placeholder="Enter surname" required="">

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Email</label>
            <div class="col-sm-8 pt-1">
                <input type="email" class="form-control personal" name="email2" placeholder="Enter email address" required="">

            </div>
        </div>
    </div>
    <div class="row">
        <div class="form-check form-check-inline">
            <label class="required" for="exampleInputEmail">Phone number</label>
            <div class="col-sm-8 pt-1">
                <input type="number" class="form-control" placeholder="Enter mobile number(preferred)" name="phone" required="">
            </div>
        </div>
    </div>
    <br>
    </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 3 Should render the employmentDetails HTML section correctly when called with a single applicant 1`] = `
"<div id="employment"><h4>Applicant 1:</h4>
        <div class="row">
        <h7 class="required">Are you employed?</h7>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed1" id="employed1" required="">
        <label class="form-check-label" for="employed1">
            Yes
        </label>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed1" id="unemployed1" required="">
        <label class="form-check-label" for="unemployed1">
            No
        </label>
        </div>
        <br>
        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 3 Should render the employmentDetails HTML section correctly when called with two applicants 1`] = `
"<div id="employment"><h4>Applicant 1:</h4>
        <div class="row">
        <h7 class="required">Are you employed?</h7>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed1" id="employed1" required="">
        <label class="form-check-label" for="employed1">
            Yes
        </label>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed1" id="unemployed1" required="">
        <label class="form-check-label" for="unemployed1">
            No
        </label>
        </div>
        <br>
        <h4>Applicant 2:</h4>
        <div class="row">
        <h7 class="required">Are you employed?</h7>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed2" id="employed2" required="">
        <label class="form-check-label" for="employed2">
            Yes
        </label>
        </div>
        <div class="form-check">
        <input class="form-check-input" type="radio" name="employed2" id="unemployed2" required="">
        <label class="form-check-label" for="unemployed2">
            No
        </label>
        </div>
        <br>
        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 4 Should render the incomeDetails HTML section correctly when called with a single employed applicant 1`] = `
"<div id="income"><h4>Applicant 1:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is your gross yearly salary you receive from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much in bonuses do you receive per year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the total amount you receive as a bonus from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="bonus" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">Are you paid anything else? if so tell us how much</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is any additional income you receive</small></p>
                                    <input type="number" class="form-control income" name="other" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 4 Should render the incomeDetails HTML section correctly when called with a single unemployed applicant 1`] = `
"<div id="income"><h4>Applicant 1:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the gross pay you receive, including benefits</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 4 Should render the incomeDetails HTML section correctly when called with applicant1 employed and applicant2 unemployed 1`] = `
"<div id="income"><h4>Applicant 1:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is your gross yearly salary you receive from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much in bonuses do you receive per year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the total amount you receive as a bonus from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="bonus" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">Are you paid anything else? if so tell us how much</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is any additional income you receive</small></p>
                                    <input type="number" class="form-control income" name="other" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        <h4>Applicant 2:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the gross pay you receive, including benefits</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 4 Should render the incomeDetails HTML section correctly when called with applicant1 unemployed and applicant2 employed 1`] = `
"<div id="income"><h4>Applicant 1:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the gross pay you receive, including benefits</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        <h4>Applicant 2:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is your gross yearly salary you receive from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much in bonuses do you receive per year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the total amount you receive as a bonus from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="bonus" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">Are you paid anything else? if so tell us how much</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is any additional income you receive</small></p>
                                    <input type="number" class="form-control income" name="other" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 4 Should render the incomeDetails HTML section correctly when called with two employed applicants 1`] = `
"<div id="income"><h4>Applicant 1:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is your gross yearly salary you receive from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much in bonuses do you receive per year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the total amount you receive as a bonus from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="bonus" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">Are you paid anything else? if so tell us how much</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is any additional income you receive</small></p>
                                    <input type="number" class="form-control income" name="other" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        <h4>Applicant 2:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is your gross yearly salary you receive from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much in bonuses do you receive per year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the total amount you receive as a bonus from your employer</small>
                                    </p>
                                    <input type="number" class="form-control income" name="bonus" step="any" required="">
                                </div>
                            </div>
                        </div>
                        <div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">Are you paid anything else? if so tell us how much</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is any additional income you receive</small></p>
                                    <input type="number" class="form-control income" name="other" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 4 Should render the incomeDetails HTML section correctly when called with two unemployed applicants 1`] = `
"<div id="income"><h4>Applicant 1:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the gross pay you receive, including benefits</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        <h4>Applicant 2:</h4><div class="row pt-2">
                            <div class="form-check form-check-inline">
                                <label class="required">How much do you earn each year, before tax?</label>
                                <br>
                                <div class="col-sm-8 pt-1">
                                    <p><small>This is the gross pay you receive, including benefits</small>
                                    </p>
                                    <input type="number" class="form-control income" name="yearly-salary" min="1" step="any" required="">
                                </div>
                            </div>
                        </div>
                         <br>
                        </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 5 Should render the outgoingsDetails HTML section correctly when called with a single applicant 1`] = `
"<div id="outgoings"><h4>Applicant 1:</h4>
                    <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required">How much are you paying off in credit cards?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is the total balance you currently hold on all credit cards</small>
                            </p>
                            <input type="number" class="form-control outgoings" name="credit-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much are you paying off in loans?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is for any outstanding loans e.g. car finance</small></p>
                            <input type="number" class="form-control outgoings" name="loan-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much in other expenses are you paying?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This could be for household bills such as electricity or gas</small></p>
                            <input type="number" class="form-control outgoings" name="other-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <br>
                </div>"
`;

exports[`HTML Generator Tests When calling renderHtml with a 5 Should render the outgoingsDetails HTML section correctly when called with two applicants 1`] = `
"<div id="outgoings"><h4>Applicant 1:</h4>
                    <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required">How much are you paying off in credit cards?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is the total balance you currently hold on all credit cards</small>
                            </p>
                            <input type="number" class="form-control outgoings" name="credit-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much are you paying off in loans?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is for any outstanding loans e.g. car finance</small></p>
                            <input type="number" class="form-control outgoings" name="loan-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much in other expenses are you paying?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This could be for household bills such as electricity or gas</small></p>
                            <input type="number" class="form-control outgoings" name="other-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <br>
                <h4>Applicant 2:</h4>
                    <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required">How much are you paying off in credit cards?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is the total balance you currently hold on all credit cards</small>
                            </p>
                            <input type="number" class="form-control outgoings" name="credit-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much are you paying off in loans?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This is for any outstanding loans e.g. car finance</small></p>
                            <input type="number" class="form-control outgoings" name="loan-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <div class="row pt-2">
                    <div class="form-check form-check-inline">
                        <label class="required" for="exampleInputName1">How much in other expenses are you paying?</label>
                        <br>
                        <div class="col-sm-8 pt-1">
                            <p><small>This could be for household bills such as electricity or gas</small></p>
                            <input type="number" class="form-control outgoings" name="other-debt" step="any" required="">
                        </div>
                    </div>
                </div>
                <br>
                </div>"
`;
